### PR TITLE
Replace unnecessary array copying with new unsafe bytestring factory

### PR DESCRIPTION
**Pull request by johanandren**
Not really super useful, most places were in tests, but can't hurt to save some array copies there either otoh.

### CODE REVIEW

File: akka-remote/src/main/scala/akka/remote/serialization/PrimitiveSerializers.scala [hunk 1]
```diff
@@ -112,7 +112,7 @@ class ByteStringSerializer(val system: ExtendedActorSystem) extends BaseSerializ
   }
 
   override def fromBinary(bytes: Array[Byte], manifest: Option[Class[_]]): AnyRef = {
-    ByteString(bytes)
+    ByteString.fromArrayUnsafe(bytes)
```
Comment (johanandren) [inline akka-remote/src/main/scala/akka/remote/serialization/PrimitiveSerializers.scala]: I'm pretty sure this is not a buffer, but an array allocated for the call, right?

File: akka-distributed-data/src/main/scala/akka/cluster/ddata/protobuf/ReplicatorMessageSerializer.scala [hunk 2]
```diff
@@ -268,7 +268,7 @@ class ReplicatorMessageSerializer(val system: ExtendedActorSystem)
     val status = dm.Status.parseFrom(bytes)
     Status(
       status.getEntriesList.asScala.map(e ⇒
-        e.getKey → AkkaByteString(e.getDigest.toByteArray()))(breakOut),
+        e.getKey → AkkaByteString.fromArrayUnsafe(e.getDigest.toByteArray()))(breakOut),
```
Comment (ktoso) [inline akka-distributed-data/src/main/scala/akka/cluster/ddata/protobuf/ReplicatorMessageSerializer.scala]: sure that this one is safe?

File: akka-stream-tests/src/test/scala/akka/stream/scaladsl/FramingSpec.scala [hunk 3]
```diff
@@ -217,7 +217,7 @@ class FramingSpec extends StreamSpec {
           val payload = referenceChunk.take(length)
           val offsetBytes = offset()
           val tailBytes = if (offsetBytes.length > 0) new Array[Byte](offsetBytes(0)) else Array.empty[Byte]
-          encodeComplexFrame(payload, fieldOffset, fieldLength, byteOrder, ByteString(offsetBytes), ByteString(tailBytes))
+          encodeComplexFrame(payload, fieldOffset, fieldLength, byteOrder, ByteString(offsetBytes), ByteString.fromArrayUnsafe(tailBytes))
```
Comment (ktoso) [inline akka-stream-tests/src/test/scala/akka/stream/scaladsl/FramingSpec.scala]: not sure it matters or that we should do this in tests... -1

Comment (johanandren) [inline akka-stream-tests/src/test/scala/akka/stream/scaladsl/FramingSpec.scala]: Saving some GC from the test suite can't hurt?

File: akka-distributed-data/src/main/scala/akka/cluster/ddata/protobuf/ReplicatorMessageSerializer.scala (hunk 2 above)
Comment (johanandren) [inline akka-distributed-data/src/main/scala/akka/cluster/ddata/protobuf/ReplicatorMessageSerializer.scala]: getDigest returns a protobuf byte string, and toByteArray on that allocates a new array and copies its content into it.

**Pull request closed without merging**